src/go/rpk/pkg/cli/transform/build.go

```diff
@@ -140,3 +146,88 @@ func buildRust(ctx context.Context, fs afero.Fs, cfg project.Config, extraArgs [
 	}
 	return nil
 }
+
+const watTemplate = `
+(module
+	(memory (import "js_vm" "memory") 0)
+
+	(func (export "file_length") (result i32)
+          (i32.const %d)
+        )
+
+	(func (export "get_file") (param $buffer_ptr i32)
+		;; Copy the source from data segment 0.
+		(memory.init $source
+			(local.get $buffer_ptr) ;; Memory destination
+			(i32.const 0)           ;; Start index
+			(i32.const %d)          ;; Length
+                )
+		(data.drop $source)
+        )
+	(data $source "%s"))
+`
+
+func createWatFromJavaScript(code []byte) string {
+	encoded := hex.EncodeToString(code)
+	escaped := strings.Builder{}
+	escaped.Grow(len(code) * 3)
+	for i := 0; i < len(encoded); i += 2 {
+		escaped.WriteRune('\\')
+		escaped.WriteByte(encoded[i])
+		escaped.WriteByte(encoded[i+1])
+	}
+	return fmt.Sprintf(watTemplate, len(code), len(code), escaped.String())
+}
+
+func buildJavaScript(ctx context.Context, fs afero.Fs, cfg project.Config) error {
+	npm, err := exec.LookPath("npm")
+	if err != nil {
+		return fmt.Errorf("npm is not available on $PATH, please download and install it: https://nodejs.org/")
+	}
```

Comment:
You can use [exec.ErrNotFound](https://pkg.go.dev/os/exec#ErrNotFound) to check whether is that `npm` is not in PATH or is any other strange error

Reply:
Same for init.go